typegen/src/tests/mod.rs

```diff
@@ -0,0 +1,1143 @@
+use parity_scale_codec::Compact;
+
+use pretty_assertions::assert_eq;
+use quote::quote;
+use scale_info::TypeInfo;
+use syn::parse_quote;
+
+use crate::{
+    tests::utils::{subxt_settings, Testgen},
+    typegen::settings::TypeGeneratorSettings,
+    utils::ensure_unique_type_paths,
+    DerivesRegistry, TypeSubstitutes,
+};
+
+mod utils;
+
+#[test]
+fn substitutes_and_derives() {
+    // set up settings
+    let settings = TypeGeneratorSettings::default()
+        .type_mod_name("my_types")
+        .derive_on_all([
+            parse_quote!(::parity_scale_codec::Decode),
+            parse_quote!(::parity_scale_codec::Emcode),
```

Comment:
nit: Is `e M code` a thing? I'd expect it to be `Encode`

Reply:
Oh that should be Encode, this is a typo.